### PR TITLE
Add_hw2

### PR DESCRIPTION
**Pull request by nico-nico13**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        driver.get("https://jdi-testing.github.io/jdi-light/index.html");
+        WebElement caret = driver.findElement(By.xpath("//li[contains(@class, 'uui-profile-menu')]"));
+        caret.click();
+        WebElement loginInput = driver.findElement(By.xpath("//input[@id='name']"));
+        loginInput.sendKeys("Roman");
+        WebElement passwordInput = driver.findElement(By.xpath("//input[@id='password']"));
+        passwordInput.sendKeys("Jdi1234");
+        WebElement enterButton = driver.findElement(By.xpath("//button[contains(@class, 'btn-login')]"));
+        enterButton.click();
+        // 4. Assert Username is loggined
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:28]: Let's separate logic blocks with empty line. It's difficult to read code without separators..

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+public class FirstExercise extends BaseClass {
+
+    SoftAssertions softAssert = new SoftAssertions();
+
+    @Test
+    public void mainFirstTest() {
+        // 2. Assert Browser title
+        softAssert.assertThat(driver.getTitle()).isEqualTo("Home Page");
+        // 3. Perform login
+        driver.get("https://jdi-testing.github.io/jdi-light/index.html");
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:19]: There is no need to open the same page several times.
`driver.get("https://jdi-testing.github.io/jdi-light/index.html");` is used in setUp method and two times in this test.

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        loginInput.sendKeys("Roman");
+        WebElement passwordInput = driver.findElement(By.xpath("//input[@id='password']"));
+        passwordInput.sendKeys("Jdi1234");
+        WebElement enterButton = driver.findElement(By.xpath("//button[contains(@class, 'btn-login')]"));
+        enterButton.click();
+        // 4. Assert Username is loggined
+        WebElement username = driver.findElement(By.id("user-name"));
+        softAssert.assertThat(username.getText()).isEqualTo("ROMAN IOVLEV");
+        // 5. Assert that there are 4 items on the header section are displayed, and they have proper texts
+        driver.get("https://jdi-testing.github.io/jdi-light/index.html");
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:32]: There is no need to open the same page several times.
`driver.get("https://jdi-testing.github.io/jdi-light/index.html");` is used in setUp method and two times in this test.

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        Boolean home = driver.findElement(By.xpath("//a[@href='index.html']")).isDisplayed();
+        softAssert.assertThat(home).isTrue();
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:34]: I would recommend in such cases to initialize webelement object and then assert this webelement state.
In this cases we can perform multiple assertions on same webelement
Example:
```java
WebElement home = driver.findElement(By.xpath("//a[@href='index.html']"));
softAssert.assertThat(home.isDisplayed()).isTrue();
```

File: src/test/java/com/epam/tc/hw2/ex2/SecondExercise.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        water.click();
+        WebElement wind = driver.findElement(By.xpath("//label[contains(., 'Wind')]/input"));
+        wind.click();
+        // 7. Select radio
+        WebElement selen = driver.findElement(By.xpath("//label[contains(.,'Selen')]/input"));
+        selen.click();
+        // 8. Select in dropdown
+        WebElement selectElement = driver.findElement(By.tagName("select"));
+        Select select = new Select(selectElement);
+        select.selectByIndex(3);
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw2/ex2/SecondExercise.java:45]: In this case it's better to select by visible text, such test would be more stable

**Review by akulebyakin — CHANGES_REQUESTED** [review]
Please review comments

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import com.epam.tc.hw2.BaseClass;
+import java.util.List;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+public class FirstExercise extends BaseClass {
+
+    SoftAssertions softAssert = new SoftAssertions();
```
Comment (akulebyakin) [inline src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:12]: There is no `softAssert.assertAll();` in test method. Without assertAll soft asserts won't perform checks

And please move softAssertions initialization to the test method

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java (hunk 4 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:34]: I didnt even figure out that was an option damn... thats smart

File: pom.xml [hunk 7]
```diff
@@ -4,8 +4,108 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.epam.tc</groupId>
+    <groupId>org.example</groupId>
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply pom.xml]: Changed back.

File: pom.xml [hunk 8]
```diff
@@ -4,8 +4,108 @@
...
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>${selenium.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>${webdrivermanager.version}</version>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply pom.xml:46]: Changed that

File: src/test/java/com/epam/tc/hw2/ex2/SecondExercise.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import com.epam.tc.hw2.BaseClass;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.annotations.Test;
+
+public class SecondExercise extends BaseClass {
+
+    @Test
+    public void mainSecondTest() {
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex2/SecondExercise.java]: Renamed

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+    @Test
+    public void mainFirstTest() {
```
[the remark this answers is not in the document]
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java]: Renamed

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java (hunk 1 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:33]: Added separation

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java (hunk 2 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java]: Oops, accidentally

File: src/test/java/com/epam/tc/hw2/ex2/SecondExercise.java (hunk 5 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex2/SecondExercise.java]: Changed that

File: src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java (hunk 6 above)
Comment (nico-nico13) [reply src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java:16]: Added assertAll

**Review by akulebyakin — APPROVED** [review]
approved

**Pull request merged**